src/sql/workbench/services/profiler/browser/profilerService.ts

```diff
@@ -317,6 +319,21 @@ export class ProfilerService implements IProfilerService {
 		if (fileURIs?.length === 1) {
 			const fileURI = fileURIs[0];
 
+			try {
+				const fileSize = (await fs.stat(fileURI.fsPath)).size;
+				const fileLimitSize = 1 * ByteSize.GB;
+				const fileOpenWarningSize = 100 * ByteSize.MB;
+
+				if (fileSize > fileLimitSize) {
+					this._notificationService.error(nls.localize('FileTooLarge', "The file is too large to open in profiler. The profiler can open files that are less than 1GB."));
+					return false;
+				} else if (fileSize > fileOpenWarningSize) {
+					this._notificationService.info(nls.localize('LargeFileWait', "Loading the file might take a moment, because the file is large."));
```

Comment:
What's the experience like now? Is there a loading spinner already? This notification seems unnecessary if that's the case. Generally we don't try to guess whether an operation is going to take a long time - the closest we may do to something like that is show a progress notification after a short delay saying something like "Opening file..." to show that it's working in the background.

Reply:
Its a blank profiler screen for now. No loading spinner. But I can take that as a follow up workitem.
![image](https://github.com/microsoft/azuredatastudio/assets/57200045/8fd66798-3e04-46c6-b369-82d20e89f796)

Reply:
That's unfortunate - alright then taking this for now and creating an issue to follow up post-release with for improving this seems fine. Thanks for verifying!